x/staking/types/authz.go

```diff
@@ -1,18 +1,19 @@
 package types
 
 import (
-	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
-
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	authz "github.com/cosmos/cosmos-sdk/x/authz/exported"
 )
 
+const gasCostPerIteration = uint64(1000)
```

Comment:
if we also have a x/deposit system, this seems a bit high for me. It's fine for now though.

Reply:
@robert-zaremba any thoughts on the gas amount here and in #8604 ?

Reply:
why would this need to be hardcoded?

Reply:
It doesn't. We can add a param, it's just that `x/authz` doesn't have a params subspace yet

Reply:
We should have a table of constants with gas costs to have better overview of what we are charging. Then, based on that table, audit and update the whole codebase. So, here we would charge based on the storage operations and CPU cycles. So, here it would be: `gas.ModuleStorageRead * num_bytes + somethin`. Probably it would make more sense to move the read/write gas charging to storage implementation. We already have cached layers, so we can do something like Ethereum: first read is more expensive then subsequent ones (of the same record).
Once we create x/deposit - we only update entries in table related to storage (make it cheaper), without updating all handlers.

I don't think `1000` is low - it creates new records.

Reply:
@atheeshp  - could you provide here few examples of other places in the SDK where we have similar operation and how much we charge there for gas?

Reply:
I don't think we have similar example in sdk but in store [(here)](https://github.com/cosmos/cosmos-sdk/blob/aleem/8984-authz-gas/store/gaskv/store.go#L55) and in ante handlers we are charging gas [(here)](https://github.com/cosmos/cosmos-sdk/blob/aleem/8984-authz-gas/x/auth/ante/basic.go#L95)